keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,924 @@
+Additionally, we can define the API scheme along with the [PodFailrePolicyReason (KEP-4443)](https://github.com/kubernetes/enhancements/pull/4479).
+
+#### Additional Story
+
+As a developer of CustomJob API built with Job API like JobSet, I want to implement the reconcile logic so that
+the controller can change the actions based on the reason why the job has been succeeded.
+
+So, it should be included in the `reason` field instead of `message` field since the reason field should be machine-readable.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    criteria:
+    - succeededIndexes: "0"
+      setSuccessCriteriaMetReason: "LeaderSucceeded"
+    - succeededIndexes: "1-9"
+      setSuccessCriteriaMetReason: "WorkersSucceeded"
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: job-container
+          image: job-image
+          command: ["./sample"]
+status:
+  conditions:
+  - type: "SuccessCriteriaMet"
+    status: "True"
+    reason: "LeaderSucceeded"
+```
+
+#### Job API
+
+Selecting one of the following API designs is possible, but the first, `setSuccessCriteriaMetReason` was preferred 
+during the JobSuccessPolicy alpha stage discussions. Because the second, `SuccessCriteriaMetReasonSuffix` would decrease the machine-readability 
+since we need to parse the reason by the separation, `As`.
+
+Furthermore, allowing the `reason` to have merging field responsibility wouldn't better and 
+decreasing the machine-readability would decrease the valuable that we have this reason in the `status.reason` field instead of `status.message` field.
+
+##### Set the entire reason
+
+```golang
+// SuccessPolicyCriteria describes a criterion for declaring a Job as succeeded.
+// Each criterion must have at least one of the "succeededIndexes" or "succeededCount" specified.
+type SuccessPolicyCriterion struct {
+	// SetSuccessCriteriaMetReason specifies the CamelCase reason for the "SuccessCriteriaMet" condition.
+	// Once the job meets this criterion, the specified reason is set to the "status.reason".
+	// The default value is "JobSuccessPolicy".
+	//
+	// +optional
+	SetSuccessCriteriaMetReason *string
+}
+```
+
+##### Set the suffix of the reason
+
+```golang
+// SuccessPolicyCriteria describes a criterion for declaring a Job as succeeded.
+// Each criterion must have at least one of the "succeededIndexes" or "succeededCount" specified.
+type SuccessPolicyCriterion struct {
+	// SuccessCriteriaMetReasonSuffix specifies the CamelCase suffix of the reason for the "SuccessCriteriaMet" condition.
+	// Once the job meets this criterion, "JobSuccessPolicy" and the specified suffix is combined with "As".
+	// For example, if specified suffix is "LeaderSucceeded", it is represented as "JobSuccessPolicyAsLeaderSucceeded".
+	//
+	// +optional
+	SuccessCriteriaMetReasonSuffix *string
+}
+```
+<</[UNRESOLVED]>>
```

Comment:
Where is the beginning of this tag?

Reply:
This is for `<<[UNRESOLVED @tenzen-y: Do we need to have the configurable reason?]>>`.